x-pack/plugins/security_solution/server/search_strategy/eql/base.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Logger } from 'kibana/server';
+import { ISearchStrategy } from '../../../../../../src/plugins/data/server';
+import {
+  EqlSearchStrategyRequest,
+  EqlSearchStrategyResponse,
+} from '../../../common/search_strategy/eql';
+import { EqlSearchResponse } from '../../lib/types';
+
+export const SECURITY_EQL_SEARCH_STRATEGY = 'security_eql_base';
+
+const dummyResponseFields = {
+  _shards: {
+    total: -1,
+    successful: -1,
+    failed: -1,
+    skipped: -1,
+  },
+  hits: {
+    max_score: -1,
+    total: -1,
+    hits: [],
+  },
+};
+
+export const securityEqlSearchStrategyProvider = (
+  logger: Logger
+): ISearchStrategy<EqlSearchStrategyRequest, EqlSearchStrategyResponse> => {
+  return {
+    search: async (context, request, options) => {
+      logger.debug(`_eql/search ${request.params?.index}`);
+
+      try {
+        const promise = context.core.elasticsearch.client.asCurrentUser.eql.search<
+          EqlSearchResponse<unknown>
+        >(request.params, request.options);
+        // Temporary workaround until https://github.com/elastic/elasticsearch-js/issues/1297
+        if (options?.abortSignal)
+          options.abortSignal.addEventListener('abort', () => promise.abort());
+        const response = await promise;
+
+        return {
+          rawResponse: {
+            ...response.body,
+            ...dummyResponseFields,
+          },
+          rawEqlResponse: response.body,
```

Comment:
```suggestion
          rawResponse: response.body
```

Applying the above change should reveal the incompatibilities between the expected `SearchResponse` type and our (currently unofficial) `EqlSearchResponse` type. To name a few, EQL responses are missing `_shards`, and `hits` has no `max_score`. Additionally, `hits` does not have its own `hits` field, but instead has either `events` or `sequences` depending on the query's syntax.

I've temporarily satisfied typescript by adding these dummy fields to satisfy [`IEsSearchResponse['rawResponse']`](https://github.com/elastic/kibana/blob/master/src/plugins/data/common/search/es_search/types.ts#L54), and added another field to hold our "true" EQL response.

We could address this in a few ways: loosen `IEsSearchResponse` to require the minimal subset of fields (`took`, `timed_out`, etc.), give it a conditional type, or diverge from `ISearchStrategy` entirely. Guidance on a path forward here would be greatly appreciated.